index.js

```diff
@@ -0,0 +1,121 @@
+window.onload = function () {
+    const canvas = document.getElementById("canvas");
+    const context = canvas.getContext("2d");
+    const BB = canvas.getBoundingClientRect();
+
+    canvas.width = window.innerWidth * 0.8;
+    canvas.height = window.innerHeight * 0.8;
+
+    const offsetX = BB.left;
+    const offsetY = BB.top;
+    const canvasWidth = canvas.width;
+    const canvasHeight = canvas.height;
+
+    let collidedList = [];
+    let nowDragging = null;
+    let rectangles = [];
+    let y = N;
+
+    for (let i = 0; i < rectanglesList.length; i++) {
```

Comment:
`rectanglesList.forEach((rectangle) => {})` or `for (let rectangle of rectanglesList) {}`? Personally I think that you should prefer `forEach` in most cases, for example `draw` function will look very clear with it: `rectangles.forEach((rectangle) => rectangle.draw(context))`